### PR TITLE
Improve linter conf

### PR DESCRIPTION
**Pull request by VincentLanglet**
POC https://github.com/sonata-project/SonataAdminBundle/pull/7903

### CODE REVIEW

File: templates/project/.php-cs-fixer.dist.php [hunk 1]
```diff
@@ -27,44 +27,31 @@
...
 
 $rules = [
-    '@PHP70Migration' => true,
-    '@PHP70Migration:risky' => true,
-    '@PHP71Migration' => true,
-    '@PHP71Migration:risky' => true,
-    '@PHPUnit57Migration:risky' => true,
-    '@PHPUnit60Migration:risky' => true,
+    '@PHP74Migration' => true,
+    '@PHP74Migration:risky' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:31]: 7.4 migration include all the previous one.

File: templates/project/.php-cs-fixer.dist.php [hunk 2]
```diff
@@ -27,44 +27,31 @@
...
+    '@PSR12' => true,
+    '@PSR12:risky' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:33]: Let's adding PSR12

File: templates/project/.php-cs-fixer.dist.php [hunk 3]
```diff
@@ -27,44 +27,31 @@
 HEADER;
 
 $rules = [
-    '@PHP70Migration' => true,
-    '@PHP70Migration:risky' => true,
-    '@PHP71Migration' => true,
-    '@PHP71Migration:risky' => true,
-    '@PHPUnit57Migration:risky' => true,
-    '@PHPUnit60Migration:risky' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:35]: Not relevant anymore

File: templates/project/.php-cs-fixer.dist.php [hunk 4]
```diff
@@ -27,44 +27,31 @@
...
+    '@PHP74Migration' => true,
+    '@PHP74Migration:risky' => true,
+    '@PSR12' => true,
+    '@PSR12:risky' => true,
     '@Symfony' => true,
     '@Symfony:risky' => true,
-    'array_syntax' => ['syntax' => 'short'],
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:38]: Include in Symfony

File: templates/project/.php-cs-fixer.dist.php [hunk 5]
```diff
@@ -27,44 +27,31 @@
...
     'combine_consecutive_issets' => true,
     'combine_consecutive_unsets' => true,
-    'compact_nullable_typehint' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:41]: Include in Symfony

File: templates/project/.php-cs-fixer.dist.php [hunk 6]
```diff
@@ -27,44 +27,31 @@
...
     'global_namespace_import' => ['import_classes' => false, 'import_constants' => false, 'import_functions' => false],
     'header_comment' => ['header' => $header],
     'list_syntax' => ['syntax' => 'short'],
-    'logical_operators' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:45]: Include in Symfony

File: templates/project/.php-cs-fixer.dist.php [hunk 7]
```diff
@@ -27,44 +27,31 @@
...
     'method_argument_space' => ['on_multiline' => 'ensure_fully_multiline'],
     'multiline_whitespace_before_semicolons' => ['strategy' => 'no_multi_line'],
-    'no_extra_blank_lines' => true,
-    'no_php4_constructor' => true,
-    'no_superfluous_phpdoc_tags' => ['allow_mixed' => true],
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:50]: Include in Symfony

File: templates/project/.php-cs-fixer.dist.php [hunk 8]
```diff
@@ -27,44 +27,31 @@
...
     'no_useless_else' => true,
     'no_useless_return' => true,
     'nullable_type_declaration_for_default_null_value' => ['use_nullable_type_declaration' => true],
     'ordered_class_elements' => true,
     'ordered_imports' => ['sort_algorithm' => 'alpha', 'imports_order' => ['class', 'function', 'const']],
-    'php_unit_method_casing' => false,
-    'php_unit_set_up_tear_down_visibility' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:57]: Include in Symfony

In symfony there is `'php_unit_method_casing' => true,`
Looking at our test, we should use it.

File: templates/project/.php-cs-fixer.dist.php [hunk 9]
```diff
@@ -27,44 +27,31 @@
...
     'php_unit_strict' => true,
-    'php_unit_test_annotation' => false,
     'php_unit_test_case_static_method_calls' => true,
-    'phpdoc_order' => true,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:61]: Include in Symfony

File: templates/project/.php-cs-fixer.dist.php [hunk 10]
```diff
@@ -27,44 +27,31 @@
...
     'phpdoc_to_comment' => ['ignored_tags' => ['psalm-suppress']],
     'single_line_throw' => false,
     'static_lambda' => true,
     'strict_comparison' => true,
     'strict_param' => true,
-    'void_return' => false,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:67]: Include in Symfony

File: templates/project/.php-cs-fixer.dist.php [hunk 11]
```diff
@@ -27,44 +27,31 @@
...
-    'no_superfluous_phpdoc_tags' => ['allow_mixed' => true],
     'no_useless_else' => true,
     'no_useless_return' => true,
     'nullable_type_declaration_for_default_null_value' => ['use_nullable_type_declaration' => true],
     'ordered_class_elements' => true,
     'ordered_imports' => ['sort_algorithm' => 'alpha', 'imports_order' => ['class', 'function', 'const']],
-    'php_unit_method_casing' => false,
-    'php_unit_set_up_tear_down_visibility' => true,
     'php_unit_strict' => true,
-    'php_unit_test_annotation' => false,
```
Comment (VincentLanglet) [inline templates/project/.php-cs-fixer.dist.php:59]: Include in Symfony with true. It remove `@test`.
Looking at our test we should use it.

**Review by jordisala1991 — APPROVED** [review]

**Pull request merged**